src/daft-io/src/lib.rs

```diff
@@ -287,6 +289,28 @@ impl IOClient {
             .await
     }
 
+    pub async fn get_multipart_writer(
```

Comment:
Another approach is add a `async fn create_multipart_writer(&self, uri: &str) -> Result<Option<Box<dyn MultipartWriter>>>` into `ObjectSource`, and each object source implement this API if the object source support mpu, the pros is the implementation can encapsulate the create logical internally, e.g. fetch part size, concurrency, the cons is it's a little bit wried that the object source has a create_mpu_writer behaivor, so I keep a factory method here.

Reply:
I think I would actually prefer that approach because that would centralize the logic to get/create an object source to `IOClient::get_source_and_path`. 

For example, in the below code we are creating an `S3LikeSource` from scratch where there is almost certainly one already created in `self.source_type_to_store` already. Having this additional method on `IOClient` just makes the potential for something like this to accidentally happen higher.

Reply:
sure, thanks man, will do.